bridge_ui/src/components/Transactions.tsx

```diff
@@ -0,0 +1,350 @@
+  },
+  nested: {
+    paddingLeft: theme.spacing(4),
+  },
+  loader: {
+    marginLeft: theme.spacing(50)
+  },
+  error: {
+    marginTop: theme.spacing(1),
+    textAlign: "center",
+  },
+}));
+
+async function getTxsByStatus(
+  status: TxStatus,
+  sourceChainId: ChainId,
+  targetChainId: ChainId
+): Promise<Transaction[]> {
+  return transactionDB.txs
+    .where({
+      "status": status,
+      "sourceChainId": sourceChainId,
+      "targetChainId": targetChainId
+    })
+    .toArray()
+}
+
+async function getTxs(
+  nodeProvider: NodeProvider,
+  fromAddress: string,
+  targetChainId: ChainId,
+  txInfos: {txId: string, sequence: string}[],
+  isTransferCompleted: (sequence: string) => Promise<boolean>,
+): Promise<Transaction[]> {
+  const txsStatus = await Promise.all(
+    txInfos.map(tx => nodeProvider.transactions.getTransactionsStatus({txId: tx.txId}))
+  )
+  const txs: Transaction[] = []
+  const toTransaction = (txId: string, sequence: string, status: TxStatus): Transaction => {
+    return new Transaction(txId, fromAddress, CHAIN_ID_ALEPHIUM, targetChainId, sequence, status)
+  }
+
+  for (let index = 0; index < txInfos.length; index++) {
+    const tx = txInfos[index]
+    const status = txsStatus[index]
+    if (isAlphTxNotFound(status)) {
+      continue
+    }
+
+    if (isAlphTxConfirmed(status) && status.chainConfirmations < ALEPHIUM_MINIMAL_CONSISTENCY_LEVEL) {
+      txs.push(toTransaction(tx.txId, tx.sequence, "Pending"))
+      continue
+    }
+
+    const isCompleted = await isTransferCompleted(tx.sequence)
+    if (isCompleted) {
+      txs.push(toTransaction(tx.txId, tx.sequence, "Completed"))
+    } else {
+      txs.push(toTransaction(tx.txId, tx.sequence, "Confirmed"))
+    }
+  }
+
+  return txs
+}
+
+async function updateTxDB(
+  nodeProvider: NodeProvider,
+  fromAddress: string,
+  targetChainId: ChainId,
+  isTransferCompleted: (sequence: string) => Promise<boolean>
+) {
+  const count = await transactionDB.txs.count()
+  if (count === 0) {
+    const response = await getAlephiumTxsWithRetry(WORMHOLE_RPC_HOSTS, fromAddress, targetChainId)
+    const txs = await getTxs(
+      nodeProvider,
+      fromAddress,
+      targetChainId,
+      response.txs,
+      isTransferCompleted
+    )
+    transactionDB.transaction("rw", transactionDB.txs, () => {
+      transactionDB.txs.bulkPut(txs)
+    })
+  } else {
+    updateTxStatus(nodeProvider, targetChainId)
+  }
+}
+
+async function updateTxStatus(nodeProvider: NodeProvider, targetChainId: ChainId) {
+  const pendingTxs = await getTxsByStatus("Pending", CHAIN_ID_ALEPHIUM, targetChainId)
+  const pendingTxsStatus = await Promise.all(
+    pendingTxs.map((tx) => nodeProvider.transactions.getTransactionsStatus({txId: tx.txId}))
+  )
+  const removedTxs: string[] = []
+  const confirmedTxs: Transaction[] = []
+  pendingTxs.forEach((tx, index) => {
+    const txStatus = pendingTxsStatus[index]
+    if (isAlphTxNotFound(txStatus)) {
```

Comment:
Yes, I'll fix this, thanks!

Reply:
Sorry, I misunderstood this, we don't need to set the "Completed" status here because the data source is the indexeddb. This means that these confirmed transactions have not been redeemed, which will be marked as "Completed" after redeeming, and we only need to update the pending transactions.

Reply:
Got it, thanks! :+1: